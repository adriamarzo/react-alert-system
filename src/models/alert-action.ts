export enum AlertAction {
  Add = "add",
  Remove = "remove",
}
