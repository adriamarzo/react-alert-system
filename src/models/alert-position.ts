export enum AlertPosition {
  BottomLeft,
  BottomRight,
  TopLeft,
  TopRight,
}
