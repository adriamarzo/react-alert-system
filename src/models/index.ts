export { Id, Alert, AlertEvent, AlertEventData } from "./alert";
export { AlertAction } from "./alert-action";
export { AlertType } from "./alert-type";
export { AlertPosition } from "./alert-position";
