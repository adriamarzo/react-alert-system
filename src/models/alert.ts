import { AlertAction } from "./alert-action";
import { AlertType } from "./alert-type";

export type Id = string;
export interface Alert {
  id: Id;
  duration?: number;
  payload?: any;
  preventDuplicated?: boolean;
  type?: AlertType;
  uid?: number;
}

export interface AlertEvent extends Alert {
  action: AlertAction;
  uid: number;
}

export interface AlertEventData {
  detail: AlertEvent;
}
