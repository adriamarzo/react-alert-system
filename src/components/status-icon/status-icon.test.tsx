import React from "react";
import { render } from "@testing-library/react";
import StatusIcon from "./status-icon";
import { AlertType } from "../../models";

describe("StatusIcon component", () => {
  it("should render an info icon", () => {
    const component = render(<StatusIcon type={AlertType.Info} />);
    expect(component.container.innerHTML).toMatchSnapshot();
  });

  it("should render an warning icon", () => {
    const component = render(<StatusIcon type={AlertType.Warning} />);
    expect(component.container.innerHTML).toMatchSnapshot();
  });

  it("should render an success icon", () => {
    const component = render(<StatusIcon type={AlertType.Success} />);
    expect(component.container.innerHTML).toMatchSnapshot();
  });

  it("should render an error icon", () => {
    const component = render(<StatusIcon type={AlertType.Error} />);
    expect(component.container.innerHTML).toMatchSnapshot();
  });
});
