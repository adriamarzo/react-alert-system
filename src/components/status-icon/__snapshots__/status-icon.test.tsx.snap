// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIcon component should render an error icon 1`] = `"<svg data-testid=\\"status-icon-error\\" height=\\"24\\" width=\\"24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zM9.053 7.65l.094.083L12 10.585l2.853-2.852a1 1 0 011.497 1.32l-.083.094L13.414 12l2.86 2.861a1 1 0 01-1.32 1.497l-.094-.083-2.86-2.86-2.853 2.853a1 1 0 01-1.497-1.32l.083-.094L10.585 12 7.733 9.147a1 1 0 011.32-1.497z\\" fill=\\"#ffffff\\" fill-rule=\\"nonzero\\"></path></svg>"`;

exports[`StatusIcon component should render an info icon 1`] = `"<svg data-testid=\\"status-icon-info\\" height=\\"24\\" width=\\"24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm0 7a1 1 0 01.993.883L13 12v4.467a1 1 0 01-1.993.116L11 16.467V12a1 1 0 011-1zm0-4.5a1.226 1.226 0 110 2.452A1.226 1.226 0 0112 6.5z\\" fill=\\"#ffffff\\" fill-rule=\\"nonzero\\"></path></svg>"`;

exports[`StatusIcon component should render an success icon 1`] = `"<svg data-testid=\\"status-icon-success\\" height=\\"24\\" width=\\"24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm4.6 4.766l.087.078a1 1 0 01.075 1.327l-.078.088-5.215 5.192a1 1 0 01-1.326.075l-.087-.077-2.74-2.745a1 1 0 011.328-1.49l.087.077 2.034 2.037 4.508-4.487a1 1 0 011.327-.075z\\" fill=\\"#ffffff\\" fill-rule=\\"nonzero\\"></path></svg>"`;

exports[`StatusIcon component should render an warning icon 1`] = `"<svg data-testid=\\"status-icon-warning\\" height=\\"24\\" width=\\"24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M13.55 3.868l.077.138 7.18 14.357a1.823 1.823 0 01-1.477 2.631l-.15.006H4.82a1.823 1.823 0 01-1.691-2.493l.064-.144 7.18-14.357a1.82 1.82 0 01.666-.73l.147-.083.13-.059a1.819 1.819 0 012.234.734zM12 4.822L4.82 19.179h14.36L12 4.822zM12 16a1 1 0 110 2 1 1 0 010-2zm0-7.172a.91.91 0 01.904.805l.006.106v3.809a.91.91 0 01-1.813.106l-.007-.106V9.739a.91.91 0 01.91-.91z\\" fill=\\"#ffffff\\" fill-rule=\\"nonzero\\"></path></svg>"`;
